lighthouse-core/test/lib/tracehouse/trace-processor-test.js

```diff
@@ -322,6 +324,31 @@ describe('TraceProcessor', () => {
       });
     });
 
+    describe('finds correct LCP', () => {
+      it('if there was a tracingStartedInPage after the frame\'s navStart', () => {
+        const trace = TraceProcessor.computeTraceOfTab(lcpTrace);
+        assert.equal(trace.mainFrameIds.frameId, '906A10385298DD996B521026AF4DA204');
+        assert.equal(trace.navigationStartEvt.ts, 1671221915754);
+        assert.equal(trace.firstContentfulPaintEvt.ts, 1671226617803);
+        assert.equal(trace.largestContentfulPaintEvt.ts, 1671236939268);
+        assert.ok(!trace.invalidLcp);
+      });
+
+      it('invalidates if last event is ::Invalidate', () => {
+        const invalidLcpTrace = createTestTrace({navigationStart: 0, traceEnd: 2000});
+        const frame = invalidLcpTrace.traceEvents[0].args.frame;
+        const args = {frame};
+        const cat = 'loading,rail,devtools.timeline';
+        invalidLcpTrace.traceEvents.push(...[
```

Comment:
could just `.push` each element as an argument instead of the spread :)

Reply:
fwiw I prefer this method over that. i like that you can "pump" the middle of the grammar to add more entires .... (the reason we all add `,` on the last items)

Reply:
> I prefer this method over that

😮 


what??

> the reason we all add , on the last items

this can be done for method calls too :)

Reply:
lol, I agree with the

> 😮
> 
> what??

even without the trailing comma, obfuscating positional arguments by spreading positioned elements directly into positional arguments seems not worth it for a test that's very unlikely to ever add more arguments :P

Reply:
fiiiine. but the real villain in this story is JS not having a `Array.prototype.pushAll` (concat does not count).

> this can be done for method calls too :)

TIL